arrows/vxl/average_frames.cxx

```diff
@@ -545,10 +545,13 @@ class average_frames::priv
     }
     else
     {
-      vil_image_view< PixType > tmp;
-      vil_image_view< double > output;
-      averager->process_frame( input, tmp, output );
-      return std::make_shared< vxl::image_container >( output );
+      // This is initially an unused argument and then the cast value we return
+      vil_image_view< PixType > placeholder;
+      vil_image_view< double > variance;
+      averager->process_frame( input, placeholder, variance );
+      // Convert the pixel format from to PixType
+      vil_convert_cast( variance, placeholder );
```

Comment:
variance may not be very meaningful if cast to a byte, unless it is also scaled up.  e.g. variance for many pixels may be small, like 0-10 grey values.  If we round that to integers it seems like we are losing a lot of information.